File: HomeWorks/CPP/PrimeFibosWithFunc.cpp
#include<iostream>
using namespace std;

int IsFibo(int num)
{
    int n1=0, n2=1, n=0, FiboCounter=2;
    if(num==0)
        return 0;
    else if(num==1)
        return 1;
    else
    {
        while(FiboCounter<=num)
        {
            n = n1+n2;
            n1 = n2;
            n2 = n;
            FiboCounter++;
        }
        return n;
    }
}

bool IsPrime(int num)
{
    if(num==1)
        return false;
    else if(num==2)
        return true;

    int i=2, PrimeCounter=0;
    bool isprime = true;
    while(i<num)
    {
        if(num%i == 0)
        {
            isprime=false;
            break;
            //PrimeCounter++;
        }
        i++;
    }

    if(isprime == true) //PrimeCounter==0
        return true;
    else
        return false;
}

//---------------------------------------

int main()
{
    int num, i=1, Counted=0;
    cout<<"Enter number of series : ", cin>>num;

    while(IsFibo(i)<=num)
    {
        if(IsPrime(IsFibo(i)))
            cout<<IsFibo(i)<<"\t";
        Counted++;
        i++;
    }
    cout<<"\n\nCounted : "<<Counted+1;  // +1 chon 0 ham hast.
    return 0;
}
//MadMad_67

File: homeworks/py/prime_fibos_with_func.py
def IsFibo(num):
    n1=0
    n2=1
    n=0
    i=2
    if(num==0):
        return 0
    elif(num==1):
        return 1
    else:
        while(i<=num):
            n=n1+n2
            n1=n2
            n2=n
            i+=1
        return n

def IsPrime(num):
    if(num==1):
        return False
    elif(num==2):
        return True
    else:
        i=2
        isprime = True
        while(i<num):
            if(num%i==0):
                isprime=False
                break
            i+=1
        
        if(isprime==True):
            return True
        else:
            return False

i=1
num = int(input("enter number of series : "))
while(IsFibo(i)<=num):
    if(IsPrime(IsFibo(i))):
        print(IsFibo(i))
    i+=1
#MadMad_43